### PR TITLE
Add SDiff command

### PR DESCRIPTION
**Pull request by gozeloglu**
* `SDiff` command implemented. `SMembers()` logic part exported as private helper
method in order to use it another method(e.g., inside of the `SDiff()` method).
* Tests added for `SDiff()`.

Signed-off-by: Gökhan Özeloğlu <gozeloglu@gmail.com>

### CODE REVIEW

**Comment by gozeloglu** [comment]
Hi @roseduan. I started to implement `SDiff` command. It's not finished yet. I'll check tomorrow. Maybe, I'll add some other tests and fix errors if exists.

**Comment by roseduan** [comment]
> Hi @roseduan. I started to implement `SDiff` command. It's not finished yet. I'll check tomorrow. Maybe, I'll add some other tests and fix errors if exists.

ok, `SMembers` command is already exists

**Comment by gozeloglu** [comment]
> > Hi @roseduan. I started to implement `SDiff` command. It's not finished yet. I'll check tomorrow. Maybe, I'll add some other tests and fix errors if exists.
> 
> ok, `SMembers` command is already exists

Yes, exists. But, if I used it inside of the `SDiff`, probably it can cause deadlock. That's why I wanted to export the logic of the `SMembers` as a private helper method.

**Comment by gozeloglu** [comment]
Hi @roseduan. It is ready to review. Could you review this PR?

File: sets.go [hunk 1]
```diff
@@ -146,6 +128,48 @@ func (db *RoseDB) SCard(key []byte) int {
...
+	if err != nil {
+		return nil, err
+	}
+	successiveSet := make(map[uint64]struct{})
+	for _, key := range keys[1:] {
+		members, err := db.sMembers(key)
+		if err != nil {
+			return nil, err
+		}
+		for _, k := range members {
```
Comment (roseduan) [inline sets.go:153]: the variable `k` seems to represent `key`,  `mem` or `value` is better here.

File: sets.go [hunk 2]
```diff
@@ -146,6 +128,48 @@ func (db *RoseDB) SCard(key []byte) int {
...
+			if _, ok := successiveSet[h]; !ok {
+				successiveSet[h] = struct{}{}
+			}
+		}
+	}
+	if len(successiveSet) == 0 {
+		return firstSet, nil
+	}
+	res := make([][]byte, 0)
+	for _, k := range firstSet {
```
Comment (roseduan) [inline sets.go:164]: similar to the previous comment.

**Comment by roseduan** [comment]
by the way, please fix the conflicts.

File: sets.go (hunk 1 above)
Comment (gozeloglu) [reply sets.go]: Done in d49fdfb.

File: sets.go (hunk 2 above)
Comment (gozeloglu) [reply sets.go]: Done in d49fdfb.

**Pull request merged**